### PR TITLE
Moving IFL Wrappers method to a new file from ptl_sevrer.py

### PR DESCRIPTION
**Pull request by birbalsain**
#### Describe Bug or Feature

Move ifl wrapper method to a new class 

#### Describe Your Change
moved ifl wrapper methods to a new file, ptl_wrappers.py

#### Attach Test and Valgrind Logs/Output

**After changes**
**No mom on server**
Description: Tests from given sources on platforms UBUNTU1804, SUSE15, UBUNTU1604, CENTOS8, RHEL8, CENTOS7, RHEL7, SLES12, SLES15, WIN2K16
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5867|3815|18|210|0|8|3579|

Description: Rerun Only Failed Tests From #5867
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5873|228|17|18|0|0|193|

Description: Rerun Only Failed Tests From #5893
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5909|35|21|12|0|0|2|

 **mom on server**
Description: Tests from given sources on platforms UBUNTU1804, SUSE15, UBUNTU1604, CENTOS8, RHEL8, CENTOS7, RHEL7, SLES12, SLES15, WIN2K16
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5866|3815|3|18|0|8|3786|

Description: Rerun Only Failed Tests From #5895
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5910|21|0|0|0|0|21|

**Before changes**
Description: Tests from given build on platforms UBUNTU1804, SUSE15, UBUNTU1604, CENTOS8, RHEL8, CENTOS7, RHEL7, SLES12, SLES15
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5568|3849|3|1|0|1|3844|

### CODE REVIEW

**Comment by birbalsain** [comment]
**before changes**
Description: Tests from given sources on platforms UBUNTU1804, SUSE15, UBUNTU1604, CENTOS8, RHEL8, CENTOS7, RHEL7, SLES12, SLES15, WIN2K16
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5915|3815|4|0|0|1|3810|

no mom on server
Description: Automated nightly regression
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5911|3815|39|11|0|1|3764|

**after changes**
Description: Tests from given sources on platforms UBUNTU1804, SUSE15, UBUNTU1604, CENTOS8, RHEL8, CENTOS7, RHEL7, SLES12, SLES15, WIN2K16
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5919|3815|4|0|0|1|3810|

no mom on server
Description: Tests from given sources on platforms UBUNTU1804, SUSE15, UBUNTU1604, CENTOS8, RHEL8, CENTOS7, RHEL7, SLES12, SLES15, WIN2K16
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5918|3815|49|88|0|1|3677|

Description: Rerun Only Failed Tests From #5918
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5935|137|34|2|0|0|101|

**Comment by subhasisb** [comment]
No activity from 25 days, any reason to keep this PR open?

**Comment by birbalsain** [comment]
@subhasisb I done some changes in this week and it is ready for review but not getting review.

File: test/fw/ptl/lib/pbs_testlib.py [hunk 1]
```diff
@@ -53,3 +53,4 @@
 from ptl.lib.ptl_resourceresv import *
 from ptl.lib.ptl_fairshare import *
 from ptl.lib.ptl_entities import *
+from ptl.lib.ptl_wrappers import *
```
Comment (kjakkali) [inline test/fw/ptl/lib/pbs_testlib.py:56]: we don't need wrapper import here

File: test/fw/ptl/lib/ptl_server.py [hunk 2]
```diff
@@ -145,216 +127,22 @@ class Server(PBSService):
...
-        """
-        Returns operating mode for calls to the PBS server.
-        Currently, two modes are supported, either the ``API``
-        or the ``CLI``. Default is ``API``
-        """
-        if (not API_OK or (self.ptl_conf['mode'] == PTL_CLI)):
-            return PTL_CLI
-        return PTL_API
-
     def set_op_mode(self, mode):
```
Comment (kjakkali) [inline test/fw/ptl/lib/ptl_server.py:146]: move it wrapper class

File: test/fw/ptl/lib/ptl_server.py [hunk 3]
```diff
@@ -145,216 +127,22 @@ class Server(PBSService):
...
-               (time.time() - self._conn_timer > self._conn_timeout))))):
-            pbs_disconnect(conn)
-            self._conn_timer = None
-            self._conn = None
+        super().__init__(name=None, attrs={}, defaults={}, pbsconf_file=None,
+                         snapmap={}, snap=None, client=None,
+                         client_pbsconf_file=None,
+                         db_access=None, stat=True)
 
     def set_connect_timeout(self, timeout=0):
```
Comment (kjakkali) [inline test/fw/ptl/lib/ptl_server.py:138]: Look for other functions also , if possible move them  new wrapper class.

File: test/fw/ptl/lib/ptl_server.py [hunk 4]
```diff
@@ -437,7 +222,6 @@ def isUp(self, max_attempts=None):
                 if op_mode == PTL_CLI:
                     self.status(SERVER, level=logging.DEBUG, logerr=False)
                 else:
-                    c = self._connect(self.hostname)
```
Comment (kjakkali) [inline test/fw/ptl/lib/ptl_server.py:440]: do we need remove this?

File: test/fw/ptl/lib/ptl_server.py [hunk 5]
```diff
@@ -4277,200 +1349,21 @@ def cleanup_jobs_and_reservations(self):
...
-                        rtype = binfo['type']
-                    if 'flag' in binfo:
-                        rflag = binfo['flag']
-                    self.resources[id] = Resource(id, rtype, rflag)
-
-            if obj is not None:
-                self.utils.update_attributes_list(obj)
-                obj.__dict__.update(binfo)
-
-    def counter(self, obj_type=None, attrib=None, id=None, extend=None,
```
Comment (kjakkali) [inline test/fw/ptl/lib/ptl_server.py:4417]: Let counter() be in server class.

File: test/fw/ptl/lib/ptl_wrappers.py [hunk 6]
```diff
@@ -0,0 +1,3329 @@
...
+        ATTR_FlatUID: 'True',
+        ATTR_DefaultChunk + ".ncpus": "1",
+    }
+
+    dflt_sched_name = 'default'
+
+    # this pattern is a bit relaxed to match common developer build numbers
+    version_tag = re.compile(r"[a-zA-Z_]*(?P<version>[\d\.]+.[\w\d\.]*)[\s]*")
+
+    actions = ExpectActions()
```
Comment (kjakkali) [inline test/fw/ptl/lib/ptl_wrappers.py:121]: move it to server class.

**Review by kjakkali — CHANGES_REQUESTED** [review]

**Comment by birbalsain** [comment]
no_mom_on_server
Description: Tests from given sources on platforms UBUNTU1804, SUSE15, UBUNTU1604, CENTOS8, RHEL8, CENTOS7, RHEL7, SLES12, SLES15, WIN2K16
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5967|3815|26|96|0|11|3682|

mom_on_server
Description: Tests from given sources on platforms UBUNTU1804, SUSE15, UBUNTU1604, CENTOS8, RHEL8, CENTOS7, RHEL7, SLES12, SLES15, WIN2K16
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|5966|3815|3|0|0|1|3811|

File: test/fw/ptl/lib/ptl_wrappers.py [hunk 7]
```diff
@@ -1382,9 +1387,8 @@ def expect(self, obj_type, attrib=None, id=None, op=EQ, attrop=PTL_AND,
             if len(newattr) == 0:
                 newattr = attrib
 
-            statlist = [self.counter(obj_type, newattr, id, extend, op=op,
-                                     attrop=attrop, level=logging.DEBUG,
-                                     runas=runas)]
+            statlist = [self._filter(obj_type, newattr, id, extend, op=op,
```
Comment (kjakkali) [inline test/fw/ptl/lib/ptl_wrappers.py:1390]: Add log level argument . make DEBUG as default value.

**Review by kjakkali — CHANGES_REQUESTED** [review]

**Comment by birbalsain** [comment]
**After Changes
mom_on_server and no_mom_on_server**
after_changes


**Run separately not started tests**
 Description: Tests from given sources on platforms UBUNTU1804, SUSE15, UBUNTU1604, CENTOS8, RHEL8, CENTOS7, RHEL7, SLES12, SLES15, WIN2K16
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15,WIN2K16
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|6066|52|0|0|0|0|52|

**Review by kjakkali — APPROVED** [review]
Changes looks good to me.

**Review by hirenvadalia — APPROVED** [review]

**Pull request merged**